src/main/kotlin/org/rust/lang/core/psi/ext/RsElement.kt

```diff
@@ -61,23 +57,15 @@ fun PsiFileSystemItem.findCargoPackage(): CargoWorkspace.Package? {
 }
 
 val RsElement.containingCargoTarget: CargoWorkspace.Target?
-    get() {
-        val ws = cargoWorkspace ?: return null
-        val root = crateRoot ?: return null
-        val file = root.contextualFile.originalFile.virtualFile ?: return null
-        return ws.findTargetByCrateRoot(
-            file.getInjectedFromIfDoctestInjection(project)?.crateRoot?.contextualFile?.originalFile?.virtualFile
-                ?: file
-        )
-    }
+    get() = (contextualFile.originalFile as? RsFile)?.cargoTarget
 
 val RsElement.containingCargoPackage: CargoWorkspace.Package? get() = containingCargoTarget?.pkg
 
 val PsiElement.isEdition2018: Boolean
     get() = contextOrSelf<RsElement>()?.containingCargoTarget?.edition == CargoWorkspace.Edition.EDITION_2018
 
 /** @see org.rust.ide.injected.RsDoctestLanguageInjector */
-private fun VirtualFile.getInjectedFromIfDoctestInjection(project: Project): RsFile? {
+fun VirtualFile.getInjectedFromIfDoctestInjection(project: Project): RsFile? {
```

Comment:
Maybe rename it somehow?
When it was a private method, it was localized in single file and everything was ok. But now it's public, so I'd prefer some more readable name. Аnd probably more detailed javadoc because I'm not sure that we will come up with a self-explained name.
Maybe `getDelegateRsFile` with something like 
```
/**
 * Returns real rust psi file in when the current one is injected.
 * For example, in case if language injection in rust doc tests
 *
 * @see org.rust.ide.injected.RsDoctestLanguageInjector
 */
```
as java doc

Reply:
Also, I feel some inconsistency in method signature- the receiver is `VirtualFile`, returning value is `PsiFile`.
Looks like we can safely convert it into
`fun PsiFileSystemItem.getInjectedFromIfDoctestInjection(project: Project): RsFile?`
because in all places when we have the corresponding psi element.
Also, it restricts area when this method can be called.
At the same time, in all calling places we need virtual file as well.
Just my thoughts

Reply:
I removed its usages from `RsElement.kt` (I found that they're redundant), moved to `RsFile.kt` and made it private again :)